transforms/transforms_test.go

```diff
@@ -362,6 +363,45 @@ func TestUniqueErrorForInvalidHeader(t *testing.T) {
 	}
 }
 
+func TestGroupBy(t *testing.T) {
+	transform := GroupBy(KeyIdentifier("group"))
+	input := []optimus.Row{
+		{"group": "1", "key": "a"},
+		{"group": "2", "key": "d"},
+		{"group": "3", "key": "g"},
+		{"group": "1", "key": 1},
+		{"group": "2", "key": 2},
+		{"group": "3", "key": 3},
+		{"group": "1", "key": 'b'},
+		{"group": "2", "key": 'e'},
+		{"group": "3", "key": 'h'},
+	}
+	expected := []optimus.Row{
+		{"1": []optimus.Row{input[0], input[3], input[6]}},
```

Comment:
Part of the reason I didn't add `GroupBy` to `optimus` when I first wrote it was that I wasn't sure what the right output was. After using it in a few places, I'm actually not very happy with this output. It means you need to do key/val iteration across a map with one key and one value, since the key is different every time. A couple other output options to consider that I think are more consistent and would be easier to work with:

```go
{"group": "1", "rows": []optimus.Row{...}}
```
```go
{"id": "1", "rows": []optimus.Row{...}}
```
```go
{"id": "1", "group": []optimus.Row{...}}
```
Would love to get @d-sparks's input here as well, since he's worked with GroupBy several times